packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -207,6 +209,46 @@ export default class CodeGen {
         return t.conditionalExpression(bindExpr, t.literal(''), t.literal(null));
     }
 
+    genSanitizedHtmlExpr(expr: t.Expression) {
+        const instance = this.innerHtmlInstances++;
+
+        this.usedLwcApis.add('sanitizeHtmlContent');
+
+        // Example input: <div lwc:inner-html={foo}>
+        // Output: $ctx._rawHtml$0 !== ($ctx._rawHtml$0 = $cmp.foo)
```

Comment:
I think it's important to add a note here explaining that this caching approach doesn't work in iterations.